File: python/problems/duplicates.py
"""
Write a function that finds duplicates in an
array and returns the duplicatre value.
"""


def findDuplicate(arr: list[int]):
    # dups = []
    # for idx, i in enumerate(arr):
    #     if i in arr[:idx]:
    #         dups.append(i)
    #     print(arr[:idx])
    return [i for idx, i in enumerate(arr) if i in arr[:idx]]


if __name__ == "__main__":
    print(findDuplicate([1, 2, 3, 2, 4, 3, 4]))

File: ts/problems/duplicates.ts
/**
 * This is a simple function to find
 * duplicates in an array
 */
function duplicates(arr: number[]): number[] {
  let dups: number[] = [];
  let fallback = new Set();

  for (let i = 0; i < arr.length; i++) {
    if (fallback.has(arr[i])) {
      dups.push(arr[i]);
    } else {
      fallback.add(arr[i]);
    }
  }
  return dups;
}

console.log(duplicates([1, 2, 3, 2, 4, 3]));
